[ModuleInterface] Filter using access control and @usableFromInline

jrose-apple opened the pull request:
These are the parts of a (resilient) module that affect the public interface and ABI; everything else is uninteresting. Or at least ought to be.

jrose-apple commented:
@swift-ci Please smoke test

jrose-apple commented:
In the spirit of "review every non-obvious change so that multiple people know what's going on", tagging @harlanhaskins for review.

harlanhaskins commented:
LGTM. Think we should add a test for implcitly-public members in `public extension UFIProto` and `public extension PublicProto`?

jrose-apple commented:
Good idea, will do.

jrose-apple committed:
[ModuleInterface] Filter using access control and @usableFromInline

These are the parts of a (resilient) module that affect the public
interface and ABI; everything else is uninteresting. Or at least ought
to be.

jrose-apple commented:
Okay, `public extension UFIProto` is already rejected, but I added the other one.

@swift-ci Please smoke test
